tls/s2n_server_new_session_ticket.c

```diff
@@ -253,3 +253,58 @@ int s2n_tls13_server_nst_write(struct s2n_connection *conn, struct s2n_stuffer *
 
     return S2N_SUCCESS;
 }
+
+int s2n_tls13_server_nst_read(struct s2n_connection *conn, struct s2n_stuffer *input)
+{
+    notnull_check(conn);
+    notnull_check(input);
+
+    uint32_t ticket_lifetime = 0;
+    GUARD(s2n_stuffer_read_uint32(input, &ticket_lifetime));
+
+    struct s2n_ticket_fields ticket_fields = { 0 };
+    GUARD(s2n_stuffer_read_uint32(input, &ticket_fields.ticket_age_add));
+
+    uint8_t ticket_nonce_len = 0;
+    struct s2n_blob nonce = { 0 };
+    GUARD(s2n_stuffer_read_uint8(input, &ticket_nonce_len));
+    uint8_t nonce_data[UINT8_MAX] = { 0 };
+    GUARD(s2n_blob_init(&nonce, nonce_data, ticket_nonce_len));
+    GUARD(s2n_stuffer_read_bytes(input, nonce.data, ticket_nonce_len));
+
+    uint16_t session_ticket_len = 0;
+    GUARD(s2n_stuffer_read_uint16(input, &session_ticket_len));
+    POSIX_ENSURE(session_ticket_len > 0, S2N_ERR_SAFETY);
+    GUARD(s2n_realloc(&conn->client_ticket, session_ticket_len));
+    GUARD(s2n_stuffer_read(input, &conn->client_ticket));
+
+    if(conn->config->session_ticket_cb != NULL) {
```

Comment:
Could we skip this entire method if the callback isn't set? That would save us a realloc and UINT8_MAX of stack memory.

Also nit: space after if